### ISSUE TITLE
Claims to skip execution but doesn't

### ISSUE THREAD

**Issue by ChristianLMI** [mojohaus/buildnumber-maven-plugin#133]
I just stumbled over a log line from the buildnumber-maven-plugin that was claiming:
```
[INFO] Skipping because we are not in root module.
```
However timestamps got created as expected. This seems to be due to this line:
https://github.com/mojohaus/buildnumber-maven-plugin/blob/3e05f277a6490728c1c07cae6a2d38c5ea297de0/src/main/java/org/codehaus/mojo/build/CreateTimestampMojo.java#L93
However I'm not exactly familiar with the intend of this feature but it appears to me that it should rather be:
```
if ( !session.getCurrentProject().isExecutionRoot() && executeRootOnly )
```
and that the block then should contain a returnstatement like the one above.

**Comment by ChristianLMI** [comment]
I still think this is a bug. I had a closer look and this was introduced in https://github.com/mojohaus/buildnumber-maven-plugin/commit/762454642fcb456a1980318b2641b0879f85a541 by @khmarbaise to fix #48. Looking at the code this seems to be ineffective as the newly introduced parameter is only used for logging but does not influence execution.

**Comment by ChristianLMI** [comment]
I intended to create a PR for this, so had another look:
* It appears to me, that the mentioned commit was mostly without effect (most changes mere reformatting, some logging added).
* The added parameter `executeRootOnly` is superfluous for the `CreateTimestampMojo`, as this is anyway only calculating a timestamp once per session.
* The logging I described above is wrong but harmless.
* The issues #48 is actually not fixed.
* The documentation of `getRevisionOnlyOnce` is not in line with the code as when using the `format` parameter it is not evaluated.

**Comment by ChristianLMI** [comment]
I'd recommend to mark `executeRootOnly` as deprecated as is has no value and eventually remove it. As for the create Mojo I'd think that it would be best if `getRevisionOnlyOnce` would be evaluated in all branches, which would also fix the issue in #48.
